src/app/_components/UserHome/UserHome.tsx

```diff
@@ -42,27 +47,70 @@ const UserHome: React.FC = () => {
     setActiveTab('Gift')
   }
 
+  const getBlindBoxData = async () => {
+    const response = await fetch('/api/blindbox', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ action: 'getList', key: walletAddress}),
+    });
+    const data = await response.json();
+    setBlindBoxList(data.data);
+  }
+  
+  useEffect(() => {
+    if (activeTab === 'Gift') {
+      setSporesList(storeSporesList || []);
+    } else {
+      getBlindBoxData()
+    }
+  }, [storeSporesList, activeTab]);
+
+  useEffect(() => {
+    let type = searchParams.get('type')
```

Comment:
- It seems that type always equals to "Gift" whichever tab user is currently on.
- Missing function to set searchParam